li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaFederatedProducerImpl.java

```diff
@@ -105,6 +107,10 @@ public LongAdder numThreadsInSend() {
     public boolean closed() {
```

Comment:
Why do we need a per producer context `closed()` public API ? It looks like you are currently using it only for tests.

Reply:
It is used for LiKafkaFederatedProducerImpl as well.